enhancements/deprecated-bundles.md

```diff
@@ -25,60 +25,49 @@ status: implementable
+- OLM prevents cluster upgrades based on whether versions are deprecated in the upgraded cluster’s index
 
 ## Proposal
 
-We propose to add a command to `opm index` called `deprecate` which takes in a set of bundle images, an index, and all the same arguments as `opm index add`. The command modifies the index by removing all versions that come before the specified bundle image in the update graph (that is, until the tail of each channel). If this operation removes the head of a channel, then the channel is removed. The removal of bundles from the update graph will not modify the update graph after the deprecated version (from the deprecated channel to the head of the channel).
-
-Versions that were removed from the index will not receive updates if subscriptions still exist for them on cluster. Ideally, appropriate communication from the operator maintainer will mean that cluster admins will upgrade to the latest supported version prior to cluster upgrade. In the event that the cluster is upgraded and `Subscriptions` to terminated bundles exist on cluster, the cluster admin can manually trigger an upgrade by either:
-
-- editing the `Subscription`'s `channel` field to start receiving updates from a different channel
-- creating a custom `CatalogSource` pointing to the previous OCP index, editing the `Subscription`'s `source` field to point to the new `CatalogSource` in order to receive updates.
-
-With the new work on skip range being steppable, the above scenario may be less likely to happen.
+We propose to add a command to `opm index` called `deprecate` which takes in a set of bundle images, an index, and all the same arguments as `opm index add`. The command modifies the index by marking all versions that come before the specified bundle image in the update graph (that is, until the tail of each channel) as deprecated as well. The deprecation of bundles from the update graph will not modify the update graph after the deprecated version (from the deprecated channel to the head of the channel).
 
 A new field will be added to the operator-registry database table `operatorbundle` called `deprecated` and the bundle images specified in the command will be marked as such.
 
-The OLM resolver will be modified with a new property type called `olm.deprecated` in order to filter out any options that have this property set and disallow installs of deprecated bundles. This also means appropriate status reporting in the subscription and installplan need to be added to OLM, and alerts will be fired if a deprecated version is installed.
+The OLM resolver will be modified with a new property type called `olm.deprecated` in order to filter out any options that have this property set and disallow installs of deprecated bundles either directly or during upgrade. This also means appropriate status reporting in the subscription and installplan need to be added to OLM, and alerts will be fired if a deprecated version is installed.
```

Comment:
My only concern here is that we are adding the deprecated as a spec instead of an annotation. 
Should not make more sense it be in CSV annotation instead?

Reply:
no because that would require modifying the bundle image and republishing that bundle.

Reply:
Hi @gallettilance, 

I am not saying an index label annotation. My suggestion here is to be a CSV annotation instead such as `maxOCPVersion`

Reply:
you cannot modify the CSV after it has been published right?

Reply:
all clarified tks